samples/spring-data-jdbc/src/main/java/com/google/cloud/spanner/sample/DatabaseSeeder.java

```diff
@@ -0,0 +1,345 @@
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Component;
+import org.springframework.util.FileCopyUtils;
+
+/** This component creates the database schema and seeds it with some random test data. */
+@Component
+public class DatabaseSeeder {
+
+  /** Randomly generated names. */
+  public static final ImmutableList<Singer> INITIAL_SINGERS =
+      ImmutableList.of(
+          new Singer("Aaliyah", "Smith"),
+          new Singer("Benjamin", "Jones"),
+          new Singer("Chloe", "Brown"),
+          new Singer("David", "Williams"),
+          new Singer("Elijah", "Johnson"),
+          new Singer("Emily", "Miller"),
+          new Singer("Gabriel", "Garcia"),
+          new Singer("Hannah", "Rodriguez"),
+          new Singer("Isabella", "Hernandez"),
+          new Singer("Jacob", "Perez"));
+
+  private static final Random RANDOM = new Random();
+
+  private final JdbcTemplate jdbcTemplate;
+
+  @Value("classpath:create_schema.sql")
+  private Resource createSchemaFile;
+
+  @Value("classpath:drop_schema.sql")
+  private Resource dropSchemaFile;
+
+  /** This value is determined once using a system query, and then cached. */
+  private final Supplier<Boolean> isCloudSpannerPG;
+
+  public DatabaseSeeder(JdbcTemplate jdbcTemplate) {
+    this.jdbcTemplate = jdbcTemplate;
+    this.isCloudSpannerPG =
+        Suppliers.memoize(() -> JdbcConfiguration.isCloudSpannerPG(jdbcTemplate));
+  }
+
+  /** Reads a resource file into a string. */
+  private static String resourceAsString(Resource resource) {
+    try (Reader reader = new InputStreamReader(resource.getInputStream(), UTF_8)) {
+      return FileCopyUtils.copyToString(reader);
+    } catch (IOException e) {
+      throw new UncheckedIOException(e);
+    }
+  }
+
+  /**
+   * Returns true if this application is currently running on a Cloud Spanner PostgreSQL database,
+   * and false if it is running on an open-source PostgreSQL database.
+   */
+  private boolean isCloudSpanner() {
+    return isCloudSpannerPG.get();
+  }
+
+  /**
+   * Removes all statements that start with a 'skip_on_open_source_pg' comment if the application is
+   * running on open-source PostgreSQL. This ensures that we can use the same DDL script both on
+   * Cloud Spanner and on open-source PostgreSQL. It also removes any empty statements in the given
+   * array.
+   */
+  private String[] updateDdlStatements(String[] statements) {
+    for (int i = 0; i < statements.length; i++) {
+      if (!isCloudSpanner()) {
+        // Replace any line that starts with '/* skip_on_open_source_pg */' with an empty string.
+        statements[i] =
+            statements[i].replaceAll("(?m)^\\s*/\\*\\s*skip_on_open_source_pg\\s*\\*/.+$", "");
+      }
+      statements[i] = statements[i].trim();
+    }
+    // Remove any empty statements from the script.
+    return Arrays.stream(statements)
+        .filter(statement -> !statement.equals(""))
```

Comment:
Good point (although I changed it to `isBlank()` and removed the `.trim()` call in the loop above).

Normally we have to stick to Java 8, but this sample uses Java 17, because of the dependency on Spring Boot, so it takes a bit of time to get used to having access to such brand new features ;-)